microbench/src/main/java/io/netty/microbench/handler/ssl/AbstractSslHandlerThroughputBenchmark.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.microbench.handler.ssl;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.util.internal.ThreadLocalRandom;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.TearDown;
+
+public class AbstractSslHandlerThroughputBenchmark extends AbstractSslHandlerBenchmark {
+    @Param({ "64", "128", "512", "1024", "4096" })
+    public int messageSize;
+
+    @Param
+    public BufferType bufferType;
+
+    public enum BufferType {
+        HEAP {
+            @Override
+            ByteBuf newBuffer(int size) {
+                return PooledByteBufAllocator.DEFAULT.heapBuffer(size);
+            }
+        },
+        DIRECT {
+            @Override
+            ByteBuf newBuffer(int size) {
+                return PooledByteBufAllocator.DEFAULT.directBuffer(size);
```

Comment:
I think it would be better to create a new instance of the allocator for the benchmarks so you not re-use previous caches etc.